pkg/bisect/bisect.go

```diff
@@ -668,9 +671,31 @@ func (env *env) processResults(current *vcs.Commit, results []instance.EnvTestRe
 			env.log("run #%v: %v", i, verdict)
 		}
 	}
+	rep, others := mostFrequentReport(reports)
```

Comment:
Do we want to take 1, or all of them, or some subset?
E.g. if we have a flaky reproducer which triggered 3 BUG and 2 KASAN out of 20 tests, and we drop KASAN, it looks like we can get false negative test results later.

Reply:
That's a good question. 

In your example, the reproducer could have well triggered two bugs, one was caught by a `BUG` assertion and another one was caught by `KASAN`. In that case we're indeed likely to get 0 crashes on a bad commit and get invalid result. Or it could also well be that it's just one bug that was surfaced by two sanitizers.

The underlying problem is that we are bisecting not a bug, but a reproducer. In an ideal world we'd likely be returning a separate fixing commit for every actual code problem triggered by a repro, but now we have to just do the bisection until the last problem is gone..

> Do we want to take 1, or all of them, or some subset?

I think it can indeed make sense to extract several reference report types here.

But we also don't want to just take all report types from the initial `test()` run and leave all those sanitizers enabled. In the experiments I've run the most distracting type was `Hang`, we can e.g. only take it if it's in the majority of reports.

Reply:
Sounds good.

I think the code should assume there can be multiple crash types we want to detect. And then the actual crash selection function can have some heuristics to include/exclude some crash types.
E.g. if we have 9 BUGs + 1 KASAN, perhaps we have disable KASAN.